x-pack/plugins/ml/public/settings/calendar/list/table/table.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {
+  EuiButton,
+  EuiLink,
+  EuiInMemoryTable,
+} from '@elastic/eui';
+
+import chrome from 'ui/chrome';
+
+
+export function CalendarsListTable({
+  calendarsList,
+  onDeleteClick,
+  setSelectedCalendarList,
+  loading,
+  canCreateCalendar,
+  canDeleteCalendar,
+  mlNodesAvailable,
+  itemsSelected
+}) {
+
+  const sorting = {
+    sort: {
+      field: 'calendar_id',
+      direction: 'asc',
+    }
+  };
+
+  const pagination = {
+    initialPageSize: 20,
+    pageSizeOptions: [10, 20]
+  };
+
+  const columns = [
+    {
+      field: 'calendar_id',
+      name: 'ID',
+      sortable: true,
+      truncateText: true,
+      render: (id) => (
+        <EuiLink
+          href={`${chrome.getBasePath()}/app/ml#/settings/calendars_list/edit_calendar/${id}`}
+        >
+          {id}
+        </EuiLink>
+      )
+    },
+    {
+      field: 'job_ids_string',
+      name: 'Jobs',
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'events_length',
```

Comment:
This column is currently sorting as Strings rather than on numeric count:

![image](https://user-images.githubusercontent.com/7405507/49731875-a2a61780-fc74-11e8-99d5-4837c4855944.png)

It should now be possible to apply a custom sort on columns - see https://github.com/elastic/eui/pull/929. Possibly one for a follow-up PR as I suspect the 'item count' column on the Filter Lists table also needs a similar fix.

Reply:
I'll make a note on the meta issue for a follow-up PR 👌 Thanks, Pete!